pkg/apis/api/v1alpha1/astarte_types.go

```diff
@@ -242,6 +242,37 @@ type AstarteRabbitMQSpec struct {
 	EventsExchangeName string `json:"eventsExchangeName,omitempty"`
 }
 
+type AstarteCassandraSSLConfigurationSpec struct {
+	Enabled bool `json:"enabled"`
+	// +optional
+	CustomCASecret v1.LocalObjectReference `json:"customCASecret,omitempty"`
+	// +optional
+	SNI *bool `json:"sni,omitempty"`
+	// +optional
+	CustomSNI string `json:"customSNI,omitempty"`
+}
+
+type AstarteCassandraConnectionSecretSpec struct {
+	Name        string `json:"name"`
+	UsernameKey string `json:"usernameKey"`
+	PasswordKey string `json:"passwordKey"`
+}
+
+type AstarteCassandraConnectionSpec struct {
+	// +optional
+	Username string `json:"username,omitempty"`
+	// +optional
+	Password string `json:"password,omitempty"`
+	// +optional
+	PoolSize string `json:"poolSize,omitempty"`
+	// +optional
+	AutodiscoveryEnabled *bool `json:"autodiscoveryEnabled,omitempty"`
+	// +optional
+	SSLConfiguration AstarteCassandraSSLConfigurationSpec `json:"sslConfiguration,omitempty"`
+	// +optional
+	Secret *AstarteCassandraConnectionSecretSpec `json:"secret,omitempty"`
```

Comment:
I know you kept it consistent with RabbitMQ, but the name kinda sucks (and we should change it there, too). I would go for something a little more explicit, such as `LoginCredentialsSecret` or the likes. Proposals welcome.